cli/enginecli/test_gofp.py

```diff
@@ -1,10 +1,13 @@
 import unittest
```

Comment:
[locust](https://github.com/bugout-dev/locust) changes in this file:

Generated (from repo root) with:

```bash
locust -r . main HEAD -f json -o diffs.json
```

- - -

Top-level objects with changes in test_gofp.py:

```bash
jq '.locust[] | select(.file == "cli/enginecli/test_gofp.py") | .changes[].name' diffs.json
```

Returns:
```
"TestPlayerFlow"
"TestAdminFlow"
"TestCallbacks"
"GOFPTestCase"
"TestFullGames"
"math"
"web3"
".None.GOFPFacet"
".None.MockTerminus"
".None.MockErc20"
".None.MockERC721"
".None.GOFPPredicates"
```

(The `.None.` looks like a small bug in locust)

Reply:
`GOFPTestCase`:

```
$ jq '.locust[] | select(.file == "cli/enginecli/test_gofp.py") | .changes[] | select(.name == "GOFPTestCase")' diffs.json
{
  "name": "GOFPTestCase",
  "type": "class",
  "line": 284,
  "changed_lines": 21,
  "total_lines": 85,
  "children": [
    {
      "name": "GOFPTestCase.setUpClass",
      "type": "function",
      "line": 286,
      "changed_lines": 21,
      "total_lines": 78,
      "children": []
    },
    {
      "name": ".None.GOFPPredicates",
      "type": "usage",
      "line": 302,
      "changed_lines": 1,
      "total_lines": 1,
      "children": []
    },
    {
      "name": ".None.GOFPPredicates.GOFPPredicates",
      "type": "usage",
      "line": 302,
      "changed_lines": 1,
      "total_lines": 1,
      "children": []
    }
  ]
}
```

It's just the deployment of predicates.

Reply:
`TestCallbacks`:

```
$ jq '.locust[] | select(.file == "cli/enginecli/test_gofp.py") | .changes[] | select(.name == "TestCallbacks") | .children[] | select(.type == "function") | .name' diffs.json
"TestCallbacks.test_session_staking_predicate"
"TestCallbacks.test_path_choice_predicate"
```

These tests check that admins can set predicates, and that the predicate callbacks function as expected.

Reply:
`TestPlayerFlow`:

```
$ jq '.locust[] | select(.file == "cli/enginecli/test_gofp.py") | .changes[] | select(.name == "TestPlayerFlow") | .children[] | select(.type == "function")' diffs.json
{
  "name": "TestPlayerFlow.test_player_can_stake_and_unstake_nfts",
  "type": "function",
  "line": 1701,
  "changed_lines": 148,
  "total_lines": 148,
  "children": []
}
{
  "name": "TestPlayerFlow.test_player_can_stake_into_free_session",
  "type": "function",
  "line": 1850,
  "changed_lines": 39,
  "total_lines": 79,
  "children": []
}
{
  "name": "TestPlayerFlow.test_player_is_rewarded_for_making_a_choice_in_stages_that_have_rewards",
  "type": "function",
  "line": 2959,
  "changed_lines": 120,
  "total_lines": 140,
  "children": []
}
{
  "name": "TestPlayerFlow.test_player_is_rewarded_for_choosing_path_that_has_reward",
  "type": "function",
  "line": 3100,
  "changed_lines": 187,
  "total_lines": 187,
  "children": []
}
{
  "name": "TestPlayerFlow.test_player_can_receive_different_pool_ids_in_stage_and_path_rewards",
  "type": "function",
  "line": 3288,
  "changed_lines": 77,
  "total_lines": 148,
  "children": []
}
```

These method test path reward functionality, but not predicates.

Reply:
`TestAdminFlow`:

```
$ jq '.locust[] | select(.file == "cli/enginecli/test_gofp.py") | .changes[] | select(.name == "TestAdminFlow") | .children[] | select(.type == "function")' diffs.json
{
  "name": "TestAdminFlow.test_create_forgiving_session_then_get_session_active",
  "type": "function",
  "line": 468,
  "changed_lines": 1,
  "total_lines": 46,
  "children": []
}
{
  "name": "TestAdminFlow.test_setting_same_stage_reward_twice_overwrites_first_value",
  "type": "function",
  "line": 1393,
  "changed_lines": 35,
  "total_lines": 58,
  "children": []
}
{
  "name": "TestAdminFlow.test_set_and_get_path_rewards",
  "type": "function",
  "line": 1452,
  "changed_lines": 101,
  "total_lines": 121,
  "children": []
}
{
  "name": "TestAdminFlow.test_non_game_master_cannot_set_path_rewards",
  "type": "function",
  "line": 1574,
  "changed_lines": 52,
  "total_lines": 52,
  "children": []
}
{
  "name": "TestAdminFlow.test_setting_same_path_reward_twice_overwrites_first_value",
  "type": "function",
  "line": 1627,
  "changed_lines": 71,
  "total_lines": 71,
  "children": []
}
```

Nothing related to predicates (that stuff is in `TestCallbacks`). Tests reward flows for admins - pertinent to new path rewards.